ui/hooks/useTransactionDisplayData.js

```diff
@@ -121,18 +129,53 @@ export function useTransactionDisplayData(transactionGroup) {
   // This value is used to determine whether we should look inside txParams.data
   // to pull out and render token related information
   const isTokenCategory = TOKEN_CATEGORY_HASH[type];
-
   // these values are always instantiated because they are either
   // used by or returned from hooks. Hooks must be called at the top level,
   // so as an additional safeguard against inappropriately associating token
   // transfers, we pass an additional argument to these hooks that will be
   // false for non-token transactions. This additional argument forces the
   // hook to return null
-  const token =
-    isTokenCategory &&
-    knownTokens.find(({ address }) =>
-      isEqualCaseInsensitive(address, recipientAddress),
-    );
+  let token = null;
+  const [currentAssetDetails, setCurrentAssetDetails] = useState(null);
+
+  if (isTokenCategory) {
+    token =
+      knownTokens.find(({ address }) =>
+        isEqualCaseInsensitive(address, recipientAddress),
+      ) ||
+      detectedTokens.find(({ address }) =>
+        isEqualCaseInsensitive(address, recipientAddress),
+      ) ||
+      tokenList[recipientAddress.toLowerCase()];
+  }
+  useEffect(() => {
+    async function getAndSetAssetDetails() {
+      const assetDetails = await getAssetDetails(
+        recipientAddress,
+        senderAddress,
+        initialTransaction?.txParams?.data,
+        knownNfts,
+      );
+      setCurrentAssetDetails(assetDetails);
+    }
+    if (isTokenCategory && !token) {
+      getAndSetAssetDetails();
+    }
```

Comment:
Hey @NiranjanaBinoy : any reason for preferring above approach. May be this:

```
if (isTokenCategory && !token) {
      const assetDetails = await getAssetDetails(
        recipientAddress,
        senderAddress,
        initialTransaction?.txParams?.data,
        knownNfts,
      );
      setCurrentAssetDetails(assetDetails);
}
```

Reply:
Updated the changes